x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameTransformState.java

```diff
@@ -33,28 +34,43 @@
 public class DataFrameTransformState implements Task.Status, PersistentTaskState {
     public static final String NAME = DataFrameField.TASK_NAME;
 
-    private final IndexerState state;
+    private final DataFrameTransformTaskState state;
+    private final IndexerState indexerState;
     private final long generation;
 
     @Nullable
     private final SortedMap<String, Object> currentPosition;
+    @Nullable
+    private final String reason;
 
-    private static final ParseField STATE = new ParseField("transform_state");
+    private static final ParseField STATE = new ParseField("state");
+    private static final ParseField TRANSFORM_STATE = new ParseField("transform_state");
     private static final ParseField CURRENT_POSITION = new ParseField("current_position");
     private static final ParseField GENERATION = new ParseField("generation");
+    private static final ParseField REASON = new ParseField("reason");
 
     @SuppressWarnings("unchecked")
     public static final ConstructingObjectParser<DataFrameTransformState, Void> PARSER = new ConstructingObjectParser<>(NAME,
-            args -> new DataFrameTransformState((IndexerState) args[0], (HashMap<String, Object>) args[1], (long) args[2]));
+            args -> new DataFrameTransformState((DataFrameTransformTaskState) args[0],
+                (IndexerState) args[1],
+                (HashMap<String, Object>) args[2],
```

Comment:
The argument is just `Map`, so casting to `HashMap` here just applies an unnecessary constraint.